src/main/java/org/lightcouch/CouchDbClient.java

```diff
@@ -95,6 +98,41 @@
  */
 public class CouchDbClient extends CouchDbClientBase {
 
+    //User-Agent value for the client
+    public static final String USER_AGENT;
+
+    static {
+        String ua = "java-cloudant";
+        String version = "unknown";
+        final URL url = CouchDbClient.class.getClassLoader().getResource("client.properties");
+        final Properties properties = new Properties();
+        InputStream propStream = null;
+        try {
+            properties.load((propStream = url.openStream()));
+            ua = properties.getProperty("user.agent.name", ua);
+            version = properties.getProperty("user.agent.version", version);
+        } catch (Exception ex) {
+            //swallow exception and keep using default values
+        } finally {
+            if (propStream != null) {
+                try {
+                    propStream.close();
+                } catch (IOException e) {
+                    //can't do anything else
```

Comment:
If there is an error trying to parse the client.properties we will use a default User-Agent value of "java-cloudant/unknown ..." on the request. I don't think this value should have any impact on the HTTP response at all because a user could change the User-Agent string anyway.
Is that the response you meant?
Or do you feel we should log an error client side?

My view is that the user probably doesn't care if the User-Agent string is set correctly or not they just want the client to work so we should just do our best to set it correctly and proceed.